2_process/src/process_data.R

```diff
@@ -1,40 +1,47 @@
-# Prepare and save the data for plotting
-process_data_from_sb <- function(data_filepath, outer_dir){
-  eval_data <- readr::read_csv(data_filepath, col_types = 'iccd') %>%
-  filter(str_detect(exper_id, 'similar_[0-9]+')) %>%
-  mutate(col = case_when(
-    model_type == 'pb' ~ '#1b9e77',
-    model_type == 'dl' ~'#d95f02',
-    model_type == 'pgdl' ~ '#7570b3'
-  ), pch = case_when(
-    model_type == 'pb' ~ 21,
-    model_type == 'dl' ~ 22,
-    model_type == 'pgdl' ~ 23
-  ), n_prof = as.numeric(str_extract(exper_id, '[0-9]+')))
-  
-  # Save the processed data
-  readr::write_csv(eval_data, file = file.path(outer_dir, '2_process', 'out', 'model_summary_results.csv'))
-  
+# Prepare the data for plotting
+process_data <- function(in_filepath) {
+  eval_data <- readr::read_csv(in_filepath, col_types = "iccd") %>%
+    filter(str_detect(exper_id, "similar_[0-9]+")) %>%
+    mutate(col = case_when(
+      model_type == "pb" ~ "#1b9e77",
+      model_type == "dl" ~ "#d95f02",
+      model_type == "pgdl" ~ "#7570b3"
+    ), pch = case_when(
+      model_type == "pb" ~ 21,
+      model_type == "dl" ~ 22,
+      model_type == "pgdl" ~ 23
+    ), n_prof = as.numeric(str_extract(exper_id, "[0-9]+")))
+
   return(eval_data)
 }
 
+# Save the processed data
+write_csv <- function(out_filepath, data) {
+  readr::write_csv(data, file = out_filepath)
+
+  return(out_filepath)
+}
```

Comment:
I would avoid giving this the same name as `readr::write_csv()`. That is the most commonly-used CSV writing function, and is usually used without the full package specification (i.e. people just call it as `write_csv()`). Since people expect its output to be the same as its input, you might introduce a confusing namespace conflict.